vernac/declaremods.ml

```diff
@@ -301,23 +301,27 @@ and load_keep i ((sp,kn),kobjs) =
 let mark_object f obj (exports,acc) =
   (exports, (f,obj)::acc)
 
-let rec collect_module_objects (f,mp) acc =
+let rec collect_module_objects_rev acc (f,mp) =
   (* May raise Not_found for unknown module and for functors *)
   let modobjs = ModObjs.get mp in
   let prefix = modobjs.module_prefix in
-  let acc = collect_objects f 1 prefix modobjs.module_keep_objects acc in
-  collect_objects f 1 prefix modobjs.module_substituted_objects acc
+  let acc = collect_objects_rev f 1 prefix acc modobjs.module_keep_objects in
+  collect_objects_rev f 1 prefix acc modobjs.module_substituted_objects
```

Comment:
If you want to produce the rev order I think you need to swap the order of processing for keep and subst objects.
I don't think there's a real impact there though as they shouldn't interfere.

Reply:
I fixed this, but as expected this does not change the behavior.